CMakeLists.txt

```diff
@@ -121,7 +121,7 @@ else()
       ${RAFT_DIR}
   )
 
-  target_compile_features(triton-fil-backend PRIVATE cxx_std_11)
+  target_compile_features(triton-fil-backend PRIVATE cxx_std_17)
```

Comment:
Can we use `set(CMAKE_CXX_STANDARD 17)` instead?

Reply:
In general modern CMake encourages setting these sort of requirements for each target rather than project wide. You can find this guidance in most discussions of modern CMake, but I'm particularly fond of [this one](https://www.youtube.com/watch?v=y7ndUhdQuU8), which lays out the advantages clearly.

In truth, the most highly recommended practice would be to more narrowly specify the features we need so that a compiler that supports just those features but not all of C++17 would be able to compile it, but I find it fairly annoying to remember exactly which features I need to enable, so I generally just enable the entire standard.